walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -13,11 +13,111 @@
 // limitations under the License.
 
 package com.google.sps;
-
 import java.util.Collection;
+import java.util.ArrayList;
+import java.util.Set;
+import java.util.HashSet;
 
 public final class FindMeetingQuery {
+  private static final int TIME_INCREMENT = 1;
+
+  /**
+   * Return a collection of TimeRange objects representing the available times when
+   * all attendees from request are available to meet.
+   * events is a collect of Events of when at least one attendee is busy. The Events
+   * in events can be considered busy times when the meeting can't be scheduled.
+   */
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    if (request.getDuration() > TimeRange.WHOLE_DAY.duration()) {
+      return new ArrayList<TimeRange>();
+    }
+
+    Collection<TimeRange> meetingOptions;
+    meetingOptions = busyToFree(events, request);
+    meetingOptions = condenseTimeRanges(meetingOptions);
+    return meetingOptions;
+  }
+
+
+  /**
+   * Given a Collection of Events that represent when a user is unavailable, return a Collection of
+   * Events representing when a user is available. Each event in the returned Collection will have
+   * at least one attendee that can also be found in meetingRequest.getAttendees(). The start time
+   * of Events in the Collection returned whose start times are adjacent to each other will be
+   * separated by a time interval of meetingRequest.getDuration().
+   */
+  private static Collection<TimeRange> busyToFree(Collection<Event> busyEvents, MeetingRequest meetingRequest) {
+    Collection<TimeRange> availableTimeSlots = new ArrayList<TimeRange>(busyEvents.size() / 2);
+    final int eventDuration = (int)meetingRequest.getDuration();
+
+    // For every timeslot of a day,
+    for (int meetingTime = TimeRange.START_OF_DAY; meetingTime <= TimeRange.END_OF_DAY;
+        meetingTime += TIME_INCREMENT) {
+      // see if this time slot is a free time slot.
+      boolean allCanAttend = true;
+      TimeRange potentialMeeting = TimeRange.fromStartDuration(meetingTime, eventDuration);
+      for (Event event : busyEvents) {
+        Set<String> meetingAttendees = new HashSet<String>(meetingRequest.getAttendees());
+        meetingAttendees.retainAll(event.getAttendees()); // meetingAttendees ⋂ event
```

Comment:
A line by line is not needed. Just a small comment above a chunk of code and what it does is sufficient.